pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
```

Comment:
> Why "atomic"? Is that the name of the "kind" of signature you have?

[Quoting](https://github.com/openshift/origin/pull/8371#issuecomment-206383470) @mtrmac:

> Re: type, I’d suggest atomic rather than saying gpg; if at all possible we want to steer users to using the provided tools (and especially the future provided policy evaluation engine) instead of suggesting they should be parsing the contents manually; in fact this ability to provide metadata without users having to parse the contents is one of the primary reasons for having an ImageSignature object instead of a raw blob.

As for the version. If we find the `atomic` kind unsufficient in the future, we can either change the name completely or append a version suffix if the next format will be similar enough. The combination of `Type` and `Version` should map to a particular signature details structure and to a particular parsing function. I don't find it necessary to have two fields for this purpose instead of one. But I don't have a strong opinion on this one. @mtrmac what's your opinion on a version field inside the signature?